server/src/repository/manage.rs

```diff
@@ -1,36 +1,77 @@
-use std::path::Path;
-use std::process::{Stdio};
-use anyhow::anyhow;
+use std::os::unix;
+use std::path::{Path, PathBuf};
+use anyhow::{anyhow, Context};
 use tokio::process::Command;
+use crate::repository::crypto;
 
 fn db_file(name: &str) -> String {
     format!("{name}.db.tar.gz")
 }
 
+fn old_path(path: &Path) -> PathBuf {
+    path.with_file_name(format!("{}.old", path.file_name().unwrap_or_default().to_str().unwrap_or_default()))
+}
+
+fn sig_path(path: &Path) -> PathBuf {
+    path.with_file_name(format!("{}.sig", path.file_name().unwrap_or_default().to_str().unwrap_or_default()))
+}
+
+fn sign_repository(name: &str, dir: &Path) -> anyhow::Result<()> {
+    let db_path = &dir.join(format!("{name}.db"));
+    let db_archive_path = &dir.join(db_file(name));
```

Comment:
why does this have a function and the others are format?, I think a format would suffice here